### PR TITLE
[cyrus-sasl,krb5] Fix dependencies

### PR DESCRIPTION
**Pull request by dg0yt**
krb5: Export pthread linkage requirement
cryus-sasl:
- Actually use the provided deps on x64-linux, using pkg-config to get usage requirements.
- Switch to lmdb instead of berkeleydb, available on all platforms.
- Improve nmake configuration.
- Install pc files for windows.

### CODE REVIEW

**Comment by timkpaine** [comment]
@dg0yt I can get it sorted in https://github.com/microsoft/vcpkg/pull/39894, don't stress too much 😅

**Comment by dg0yt** [comment]
Linux is broken, too: No db support and no kerberos. And port krb5 shows another issue...

I'm on a good way, but I will switch from berkeley-db to lmdb which is maintained and supported across platforms.

**Comment by timkpaine** [comment]
I had to do this locally, idk why homebrew isn't putting `pkg.m4` in the correct place:

```bash
cp -r /opt/homebrew/Cellar/pkg-config/0.29.2_3/share/aclocal/pkg.m4 /opt/homebrew/share/aclocal/
```

File: ports/cyrus-sasl/portfile.cmake [hunk 1]
```diff
@@ -7,40 +7,60 @@ vcpkg_download_distfile(ARCHIVE
...
     vcpkg_configure_make(
         SOURCE_PATH "${SOURCE_PATH}"
-        DETERMINE_BUILD_TRIPLET
+        AUTOCONFIG
         OPTIONS
-            "--with-gssapi"
-            "--disable-macos-framework"
+            --with-dblib=lmdb
+            --with-gss_impl=mit
+            --disable-macos-framework
```
Comment (timkpaine) [inline ports/cyrus-sasl/portfile.cmake:59]: do we need a `--with-openssl=${CURRENT_INSTALLED_DIR_NATIVE}`?

**Comment by dg0yt** [comment]
The libspatialite regression is from a bug in (my code in) port curl. The fix is now included in #40081. I didn't merge it here yet because curl requires CI to build more ports, and I want more cyrus-sasl CI logs first.

File: ports/cyrus-sasl/vcpkg.json [hunk 2]
```diff
@@ -1,16 +1,14 @@
 {
   "name": "cyrus-sasl",
   "version": "2.1.28",
+  "port-version": 1,
   "description": "Cyrus SASL is an implementation of SASL that makes it easy for application developers to integrate authentication mechanisms into their application in a generic way.",
   "homepage": "https://github.com/cyrusimap/cyrus-sasl",
-  "license": "BSD-3-Clause-Attribution",
-  "supports": "linux | osx | (x64 & windows & !static & !uwp)",
+  "license": null,
```
Comment (dg0yt) [inline ports/cyrus-sasl/vcpkg.json:7]: Cf. https://tracker.debian.org/media/packages/c/cyrus-sasl2/copyright-2.1.28dfsg1-6

**Review by LilyWangLL — APPROVED** [review]

**Comment by timkpaine** [comment]
Thanks for taking care of this @dg0yt!

**Comment by timkpaine** [comment]
Sorry to pester, can this be merged? There is pending conversation on https://github.com/microsoft/vcpkg/pull/40081 which this includes.

Now that @dg0yt has cleaned up my mess and a few other messes, I really am keen to get https://github.com/microsoft/vcpkg/pull/39894 finished as this allows vcpkg-based `librdkafka` to integrate with GSSAPI-secured kafka clusters, which is the de-facto standard for any enterprise kafka cluster including those offered by e.g. Confluent, so it is really quite major functionality that is really close to being enabled 😄

**Review by BillyONeal — APPROVED** [review]
I'm not super sure how to evaluate anything about autotools changes but I don't see anything 'obviously' bogus.

**Pull request merged**